controllers/complianceeventsapi/server.go

```diff
@@ -291,46 +239,37 @@ func getPolicyForeignKey(ctx context.Context, goquDB *goqu.Database, pol Policy)
 		return key.(int), nil
 	}
 
-	foundPolicy := new(Policy)
-
-	qu := goquDB.From("policies").Where(goqu.Ex{
-		"kind":             pol.Kind,
-		"api_group":        pol.APIGroup,
-		"name":             pol.Name,
-		"spec_hash":        pol.SpecHash,
-		"namespace":        pol.Namespace,
-		"parent_policy_id": pol.ParentPolicyID,
-		"severity":         pol.Severity,
-	})
-
-	found, err := qu.ScanStructContext(ctx, foundPolicy)
-	if err != nil {
-		return 0, err
-	}
-
-	// If the row already exists
-	if found {
-		policyKeyCache.Store(polKey, foundPolicy.KeyID)
-
-		return foundPolicy.KeyID, nil
-	}
+	if pol.Spec == "" {
+		row := db.QueryRowContext(
+			ctx, fmt.Sprintf("SELECT spec FROM %s WHERE spec_hash = $1 LIMIT 1", policyTableName), pol.SpecHash,
```

Comment:
Tangent/opinion: this `Sprintf` feels weird. How much do we gain by making the table name change-able later?

I think I'd rather either have a magic string here that includes the table name, or the whole query as a constant defined elsewhere. I like that the actual query is obvious here, and a magic string would keep that nicely. But, having the whole query defined somewhere else would let us keep all of these kinds of queries defined together, possibly making it easier to review whether they are following best practices. Naming the constants might be clunky, but if they were just magic strings, I expect IDEs would be able to reveal the actual query when hovered over (VS Code does at least), which I think would otherwise be the main downside.

Reply:
Sure, I'll add it directly but I like having the query directly here than as a constant.